### PR TITLE
Typescript move.target

### PR DESCRIPTION
**Pull request by urkerab**
Also allowing the various move targets to be documented.

This found a bug in SSB!

### CODE REVIEW

File: sim/global-types.ts [hunk 1]
```diff
@@ -21,6 +21,41 @@ type BoostName = StatNameExceptHP | 'accuracy' | 'evasion';
 type BoostsTable = {[boost in BoostName]: number };
 type SparseBoostsTable = Partial<BoostsTable>;
 type Nonstandard = 'Past' | 'Future' | 'Unobtainable' | 'CAP' | 'LGPE' | 'Custom';
+/**
+ * Describes the acceptable target(s) of a move.
+ */
+type MoveTarget =
+	// Only relevant to Doubles or Triples, the move targets your ally only.
```
Comment (Zarel) [inline sim/global-types.ts]: This documentation should be in the `/** */` block, so it shows up on hover.

File: sim/global-types.ts [hunk 2]
```diff
@@ -21,6 +21,41 @@ type BoostName = StatNameExceptHP | 'accuracy' | 'evasion';
...
+	// This move targets the field or all Pokémon at once.
+	'all' |
+	// This is a spread move that also hits your ally.
+	'allAdjacent' |
+	// This is a spread move.
+	'allAdjacentFoes' |
+	// This move affects all active Pokémon on your team.
+	'allies' |
+	// This move is a side effect for your side.
+	'allySide' |
```
Comment (Zarel) [inline sim/global-types.ts]: > petition to rename
> `allySide` -> `allySideOfField`
> and
> `allyTeam` -> `alliesIncludingInactive`

File: sim/global-types.ts [hunk 3]
```diff
@@ -21,6 +21,41 @@ type BoostName = StatNameExceptHP | 'accuracy' | 'evasion';
...
+	'adjacentFoe' |
+	// This move targets the field or all Pokémon at once.
+	'all' |
+	// This is a spread move that also hits your ally.
+	'allAdjacent' |
+	// This is a spread move.
+	'allAdjacentFoes' |
+	// This move affects all active Pokémon on your team.
+	'allies' |
+	// This move is a side effect for your side.
```
Comment (Zarel) [inline sim/global-types.ts]: "adds a side condition"

File: sim/global-types.ts [hunk 4]
```diff
@@ -21,6 +21,41 @@ type BoostName = StatNameExceptHP | 'accuracy' | 'evasion';
...
+	'allySide' |
+	// This move affects all unfainted Pokémon on your team.
+	'allyTeam' |
+	// This move can hit any other active Pokémon, even in Triples.
+	'any' |
+	// This move is a side effect for your opponent.
```
Comment (Zarel) [inline sim/global-types.ts]: "adds a side condition"

File: sim/global-types.ts [hunk 5]
```diff
@@ -21,6 +21,41 @@ type BoostName = StatNameExceptHP | 'accuracy' | 'evasion';
...
+	'allAdjacent' |
+	// This is a spread move.
+	'allAdjacentFoes' |
+	// This move affects all active Pokémon on your team.
+	'allies' |
+	// This move is a side effect for your side.
+	'allySide' |
+	// This move affects all unfainted Pokémon on your team.
+	'allyTeam' |
+	// This move can hit any other active Pokémon, even in Triples.
```
Comment (Zarel) [inline sim/global-types.ts]: "not just adjacent"

File: sim/global-types.ts (hunk 1 above)
Comment (urkerab) [reply sim/global-types.ts]: I'll take your word for it. (I develop for PS on Linux so I don't use VS Code.)

**Comment by Zarel** [comment]
I mentioned on Discord:

> petition to rename
> `allySide` -> `allySideOfField`
> and
> `allyTeam` -> `alliesIncludingInactive`

your thoughts?

**Comment by urkerab** [comment]
If you're asking me, then I don't think the rename is helpful, what with the improved documentation (!)

**Comment by Zarel** [comment]
I don't think the improved documentation is easily to accessible to anyone who doesn't use VS Code or some other IDE with deep TypeScript integration.

**Comment by Zarel** [comment]
Which I'm pretty sure includes a lot of researchers like DaWob.

**Comment by Zarel** [comment]
This is a strict improvement and it's been floating too long, though (sorry, I got sick a few days ago.)

**Pull request merged**